### PR TITLE
Update for upcoming OpenFF Toolkit API breaks

### PR DESCRIPTION
**Pull request by mattwthompson**
Hi OpenMM developers,

I'm seeing some test failures in our example that uses `openmmforcefields` with our big biopolymer feature branch, probably due to `Atom.element` being removed in our upcoming API break. This change should fix that and be backwards-compatible.

### CODE REVIEW

File: devtools/conda-envs/test_env.yaml [hunk 1]
```diff
@@ -27,9 +27,11 @@ dependencies:
     # JSON caching of residue templates
   - tinydb
 
-    # openff toolkit and force fields
-  - openff-toolkit >=0.7.1
+    # OpenFF toolkit and force fields
+  - openff-toolkit >=0.10.3|<=0.10.0
   - openff-forcefields >=1.2.0
+  - pip:
+    - git+https://github.com/openforcefield/openff-toolkit.git@topology-biopolymer-refactor
```
Comment (mattwthompson) [inline devtools/conda-envs/test_env.yaml:34]: If you don't want to shoulder the complexity of testing against an unreleased version, you can grab it with a similar command locally (`pip install git+https://github.com/openforcefield/openff-toolkit.git@topology-biopolymer-refactor`) and have a look around for yourself.

Comment (mattwthompson) [reply devtools/conda-envs/test_env.yaml:34]: And for the record: I'm not suggesting this change be merged in; I want to show in the context of this PR that it should pass with the new API. I will revert this before merging if approved.

File: openmmforcefields/generators/template_generators.py [hunk 2]
```diff
@@ -236,9 +236,13 @@ def _generate_unique_atom_names(self, molecule):
...
         """
         from collections import defaultdict
+
+        # OpenFF Toolkit v0.11.0 removed Atom.element and replced it with Atom.symbol, etc.
+        uses_old_api = hasattr(molecule.atoms[0], "element")
+
         element_counts = defaultdict(int)
         for atom in molecule.atoms:
-            symbol = atom.element.symbol
+            symbol = atom.element.symbol if uses_old_api else atom.symbol
```
Comment (jchodera) [inline openmmforcefields/generators/template_generators.py:245]: Oh, that's an unexpected change.

@mattwthompson, can you point to some information about the motivation for this? Is the idea that `Atom` objects don't have to be physical elements anymore?

File: devtools/conda-envs/test_env.yaml (hunk 1 above)
Comment (jchodera) [reply devtools/conda-envs/test_env.yaml:34]: Thanks for this! We should test against both the current and preview versions, ideally. But to do that, we can't just eliminate testing the current version---I'll have to rework the test matrix.

**Comment by jchodera** [comment]
Thanks for this, @mattwthompson! 

I'll need to finish up https://github.com/openmm/openmmforcefields/pull/182 first, but then will tackle integrating this PR.

File: openmmforcefields/generators/template_generators.py (hunk 2 above)
Comment (jchodera) [reply openmmforcefields/generators/template_generators.py:245]: And if `symbol` supports non-physical elements or made-up symbols now, is there an API for telling whether the element is a real element or not?

**Comment by richardjgowers** [comment]
It would be great if this could get merged, this is breaking stuff for openfe currently

File: openmmforcefields/generators/template_generators.py (hunk 2 above)
Comment (mikemhenry) [reply openmmforcefields/generators/template_generators.py:245]: This is the PR where the change happened: https://github.com/openforcefield/openff-toolkit/pull/1182
It comes from switching to a different package.

Comment (mikemhenry) [reply openmmforcefields/generators/template_generators.py:245]: @jchodera this is the only modification needed to get openmmforcefields working with the changes in the openff-toolkit/openff-element and given they are done in a backwards compatible way, I am happy with them. If you are happy with it, I can make a new PR add this changes + tests the latest version of the openff-toolkit.

**Review by jchodera — APPROVED** [review]
Thanks!

**Review by mikemhenry — APPROVED** [review]

**Comment by mattwthompson** [comment]
Thanks!

**Comment by mattwthompson** [comment]
Unfortunately our [tests are still](https://github.com/openforcefield/openff-toolkit/runs/7742439141?check_suite_focus=true) failing; I think L966 in this loop needs a similar change:

https://github.com/openmm/openmmforcefields/blob/a59b1445d8fe41c873c052590e330e615931e782/openmmforcefields/generators/template_generators.py#L960-L967

if I had to guess, I didn't notice this previous because I was grepping around for `atom.element.symbol` or something similar - and at first glance the traceback looks like `particle` is an OpenMM object of sorts, not an OpenFF `Atom` object.

**Pull request merged**